### PR TITLE
Tighten the logic preventing step4 from applying to a master

### PR DESCRIPTION
**Pull request by jeffmccune**
Nigel points out in [1] the logic protecting the puppet master from
receiving the resources in step 4 is not as robust as it could be.  If
the PE facts are disabled for some reason or removed, then the resources
would still be applied to the puppet master.

This patch fixes the problem by explicitly checking of the
fact_is_puppetmaster fact is "false"  If the fact is removed, then this
logic will still protect the puppet master.

An agent will no longer receive the resources if their facts have been
tampered with however.  This will need to be verified on PE 1.1 and
earlier versions which may not have this fact.

[1] https://github.com/puppetlabs/puppetlabs-cve20113872/blob/master/manifests/step4.pp#L80

### CODE REVIEW

**Comment by jeffmccune** [comment]
Does anyone have any thoughts on this one?  I'm not confident these facts exist in the earlier versions of PE and I'd like to be defensive.

File: manifests/step4.pp [hunk 1]
```diff
@@ -77,7 +77,7 @@ class cve20113872::step4 {
   # Otherwise, the agent will replace the CRL and CA bundle being used
   # by the PE Apache server, causing all existing agents with an "old" certificate
   # to be unauthenticated by the master.
-  if ($fact_is_puppetmaster != 'true') {
+  if ($fact_is_puppetmaster == 'false') {
```
Comment (nigelkersten) [inline manifests/step4.pp]: As discussed in person, we could have a fact that checks for whether or not any of the artifacts exist that are only created by the migration scripts themselves when run on the master.

Comment (jeffmccune) [inline manifests/step4.pp]: I've re-written this entire patch.  It should be much more robust and backwards compatible now.  Please review again whenever you get the chance.

**Comment by jeffmccune** [comment]
I think this is ready to merge.

**Comment by nigelkersten** [comment]
This switch to case statements is totally retro. :)

**Pull request merged**